ext/phar/stream.c

```diff
@@ -325,10 +322,10 @@ static php_stream * phar_wrapper_open_url(php_stream_wrapper *wrapper, const cha
 	}
 
 	if (!PHAR_G(cwd_init) && (options & STREAM_OPEN_FOR_INCLUDE)) {
-		char *entry = idata->internal_file->filename, *cwd;
+		char *entry = ZSTR_VAL(idata->internal_file->filename), *cwd;
 
 		PHAR_G(cwd_init) = 1;
-		if ((idata->phar->is_tar || idata->phar->is_zip) && idata->internal_file->filename_len == sizeof(".phar/stub.php")-1 && !strncmp(idata->internal_file->filename, ".phar/stub.php", sizeof(".phar/stub.php")-1)) {
+		if ((idata->phar->is_tar || idata->phar->is_zip) && ZSTR_LEN(idata->internal_file->filename) == sizeof(".phar/stub.php")-1 && !strncmp(entry, ".phar/stub.php", sizeof(".phar/stub.php")-1)) {
```

Comment:
Isn't this check checking that the filename is equal to `.phar/stub.php` ? In this case better to use the zend_string_equals API IMHO